cmd/wit-bindgen-go/cmd/generate/generate.go

```diff
@@ -189,3 +194,21 @@ func writeGoPackages(packages []*gen.Package, cfg *config) error {
 	}
 	return nil
 }
+
+func writeWITPackage(res *wit.Resolve, cfg *config) error {
+	if cfg.dryRun {
+		return nil
+	}
+	witDir := filepath.Join(cfg.out, "wit")
+	if err := os.MkdirAll(witDir, cfg.outPerm); err != nil {
+		return err
+	}
+	witFilePath := filepath.Join(witDir, "webassembly.wit")
```

Comment:
Could the name should be based on the top-level WIT package, if possible, and nested in that package's directory, e.g. $out/wasi/cli/webassembly.wit?

Reply:
I think a common practice for placing WIT files is to place it at a `wit` directory. In addition, a WIT package can have multiple worlds, and `wit-bindgen-go generate` generates all of them. Take wasi:cli as an example, it generates

```
.
├── go.mod
├── go.sum
├── main.go
├── main.wasm
├── wasi
│   ├── cli
│   │   ├── stderr
│   │   │   ├── empty.s
│   │   │   └── stderr.wit.go
│   │   ├── stdin
│   │   │   ├── empty.s
│   │   │   └── stdin.wit.go
│   │   └── stdout
│   │       ├── empty.s
│   │       └── stdout.wit.go
│   ├── clocks
│   │   ├── monotonic-clock
│   │   │   ├── empty.s
│   │   │   └── monotonic-clock.wit.go
│   │   └── wall-clock
│   │       ├── empty.s
│   │       └── wall-clock.wit.go
│   ├── http
│   │   ├── incoming-handler
│   │   │   ├── empty.s
│   │   │   ├── incoming-handler.exports.go
│   │   │   └── incoming-handler.wit.go
│   │   ├── outgoing-handler
│   │   │   ├── abi.go
│   │   │   ├── empty.s
│   │   │   └── outgoing-handler.wit.go
│   │   ├── proxy
│   │   │   └── proxy.wit.go
│   │   └── types
│   │       ├── abi.go
│   │       ├── empty.s
│   │       └── types.wit.go
│   ├── io
│   │   ├── error
│   │   │   ├── empty.s
│   │   │   └── error.wit.go
│   │   ├── poll
│   │   │   ├── empty.s
│   │   │   └── poll.wit.go
│   │   └── streams
│   │       ├── abi.go
│   │       ├── empty.s
│   │       └── streams.wit.go
│   └── random
│       └── random
│           ├── empty.s
│           └── random.wit.go
└── wit
    └── webassembly.wit
```

Reply:
Multiple WIT files might be generated in the same out directory. So we need some way to avoid overwriting WIT files in the same directory

Reply:
Perhaps, we should limit `wit-bindgen-go generate` to require a World name, and the WIT file can be named `<out>/wit/<world-name>.wit`?